### PR TITLE
[SDK-545] Name change for deprecated calls

### PR DESCRIPTION
**Pull request by turtledreams**
- Changed `report_feedback` to `recordRatingWidgetWithID`. Now old call redirects to the new one.
- Changed `show_feedback_popup` to `presentRatingWidgetWithID`. Now old call redirects to the new one.
- Changed `initialize_feedback_popups` to `initializeRatingWidgets`. Now old call redirects to the new one.
- Changed `enable_feedback` to `enableRatingWidgets`. Now old call redirects to the new one.
- Changed `report_conversion` to `recordDirectAttribution`. Now old call redirects to the new one.
- Fixed some places where deprecated call was used. Changed with new one.
- Added changelog entries

### CODE REVIEW

File: lib/countly.js [hunk 1]
```diff
@@ -1832,8 +1842,25 @@
...
+        * 
+        * @deprecated use 'recordRatingWidgetWithID' in place of this call
         **/
         this.report_feedback = function(feedback) {
+            log("report_feedback, Deprecated function call! Use 'recordRatingWidgetWithID' in place of this call. Call will be redirected now!");
+            this.recordRatingWidgetWithID(feedback);
+        };
+        /**
+        * Provide information about user
+        * @param {Object} feedback - object with feedback properties
```
Comment (ArtursKadikis) [inline lib/countly.js:1854]: rename the function variable to "ratingWidget" and respectively also the name used in code comments from "feedback" to "ratingWidget".

File: lib/countly.js [hunk 2]
```diff
@@ -1858,12 +1885,21 @@
...
+        * 
+        * @deprecated use 'presentRatingWidgetWithID' in place of this call
         */
         this.show_feedback_popup = function(id) {
+            log("show_feedback_popup, Deprecated function call! Use 'presentRatingWidgetWithID' in place of this call. Call will be redirected now!");
+            this.presentRatingWidgetWithID(id);
+        };
+        /**
+        * Show specific widget popup by the widget id
+        * @param {string} id - id value of related feedback widget, you can get this value by click "Copy ID" button in row menu at "Feedback widgets" screen
```
Comment (ArtursKadikis) [inline lib/countly.js:1900]: rename "feedback widget" to "rating widget"

File: lib/countly.js [hunk 3]
```diff
@@ -1890,8 +1926,19 @@
...
         * @param {array=} enableWidgets - widget ids array
+        * 
+        * @deprecated use 'initializeRatingWidgets' in place of this call
         */
         this.initialize_feedback_popups = function(enableWidgets) {
+            log("initialize_feedback_popups, Deprecated function call! Use 'initializeRatingWidgets' in place of this call. Call will be redirected now!");
+            this.initializeRatingWidgets(enableWidgets);
+        };
+        /**
+        * Prepare widgets according current options
```
Comment (ArtursKadikis) [inline lib/countly.js:1937]: "Prepare **rating** widget"

File: lib/countly.js [hunk 4]
```diff
@@ -1954,8 +2001,19 @@
...
         * example params: {"popups":["5b21581b967c4850a7818617"]}
+        * 
+        * @deprecated use 'enableRatingWidgets' in place of this call
         **/
         this.enable_feedback = function(params) {
+            log("enable_feedback, Deprecated function call! Use 'enableRatingWidgets' in place of this call. Call will be redirected now!");
+            this.enableRatingWidgets(params);
+        };
+        /**
+        * Show feedback popup by passed widget ids array
```
Comment (ArtursKadikis) [inline lib/countly.js:2012]: "Show **rating widget** popup"

File: samples/react/src/Location-WithEffect.js [hunk 5]
```diff
@@ -14,7 +14,7 @@ const Location = (props) => {
     //So that you dont end up tracking the same view again and again
     Countly.q.push(['track_pageview', location.pathname]);
     // Initialize feedback popup by current page/pathname
```
Comment (ArtursKadikis) [inline samples/react/src/Location-WithEffect.js:16]: "Initialize **rating widget** popup"

File: lib/countly.js [hunk 6]
```diff
@@ -1832,8 +1842,25 @@
...
+        * @param {Object} feedback - object with feedback properties
+        * @param {string} feedback.widget_id - id of the widget in the dashboard
+        * @param {boolean=} feedback.contactMe - did user give consent to contact him
+        * @param {string=} feedback.platform - user's platform (will be filled if not provided)
+        * @param {string=} feedback.app_version - app's app version (will be filled if not provided)
+        * @param {number} feedback.rating - user's rating from 1 to 5
+        * @param {string=} feedback.email - user's email
+        * @param {string=} feedback.comment - user's comment
+        **/
+        this.recordRatingWidgetWithID = function(feedback) {
```
Comment (ArtursKadikis) [inline lib/countly.js:1863]: In the function bellow, replace mentions and usages of "Feedback" to "Rating widget"

**Review by ArtursKadikis — CHANGES_REQUESTED** [review]

**Comment by ArtursKadikis** [comment]
Paging dr @ar2rsawseen to check this patient out

**Comment by ar2rsawseen** [comment]
Well, the only problem I have is that some public methods are now camelCase and others are Snake_case :D 
It is part of uniting API of all SDKs?

Then you should better come up with united API first and then change all methods in one swoop, while providing backwards compatibility for web SDK cases?

But other than that, its fine

**Review by ar2rsawseen — APPROVED** [review]

**Pull request merged**